src/evergreen/api.py

```diff
@@ -148,7 +148,7 @@ def _create_session(self) -> requests.Session:
         session.mount(f"{urlparse(self._api_server).scheme}://", adapter)
         auth = self._auth
         if auth:
-            session.headers.update({"Api-User": auth.username, "Api-Key": auth.api_key})
+            session.headers.update(auth.get_auth_headers())
```

Comment:
I did a retrieval func style so it's easier to test, but I honestly don't know how to add tests or run the tests. My computer's python env is broken and I am unfamiliar with it :/